### DISCUSSION TITLE
How to set schema.default from 'none' to default

### DISCUSSION THREAD

**Discussion by dkumarkartik** [JanusGraph/janusgraph#3558] [Q&A] [answered]
**Describe the feature:**
Hello Team,
We stricter our janusGraph schema by adding two configuration property to our schema e.g schema.constraints = true
and schema.default='none'. Now it's working as expected.
Can you please let us know how can we revert it back.

**Comment by porunov** [comment, marked as answer]
`schema.default` is a `MASKABLE` property meaning you can change it for your instance.
`schema.constraints` is a `GLOBAL_OFFLINE` property meaning that you must shutdown the whole JanusGraph cluster to change this configuration (keep in mind that you need to force close any zombie instances if you have any.
See how to close zombie instances: https://github.com/JanusGraph/janusgraph/discussions/3195#discussioncomment-3595012
See configuration properties mutability explanation: https://docs.janusgraph.org/configs/configuration-reference/

**Reply by dkumarkartik** [reply]
Hello @porunov thank you for your response,
need more clarification on below questions as I am getting error while setting schema.default to blank from 'none' 
Can I have one example how can I change schema.default to anything else other than 'none'?
what are the possible value for schema.default?

**Reply by porunov** [reply]
blank option uses either a MASKABLE value if it exists or a default value if it doesn't exist. In you case you already use `none` for MASKABLE property. Thus a blank option will mean `none` in you case.
To change it to `default` you need to provide a JanusGraph configuration `schema.default=default`.
Please, see available `schema.default` options: https://docs.janusgraph.org/configs/configuration-reference/#schema

P.S. I think `schema.default` configuration is a bad name for a configuration property and instead I would call it `schema.maker` of something else, but we use that name due to historical reasons. I guess we could make an alias for this property so that people wouldn't be confused.

**Reply by MahatiPV** [reply]
Hi @porunov! Thanks for your inputs.
We are not able to apply schema.default=default when doing it through ConfiguredGraphFactory. We see the below error -
_Error in /docker-entrypoint-initdb.d/entrypoint.groovy at [12:     map.put("schema.default", default);] - startup failed:
groovysh_parse: 7: unexpected token: default @ line 7, column 31.
       map.put("schema.default", default);_
Can you please suggest how it could be resolved?

**Reply by porunov** [reply]
It looks like Groovy error and not JanusGraph error. I didn't check that but your error says: `groovysh_parse: 7: unexpected token: default @ line 7, column 31.` and shows you `map.put("schema.default", default);`. Seems `default` in this case is treated as a variable (which doesn't exist in Groovy context) and not as String (which is expected for this configuration option).
Thus, I would assume encapsulating `default` into `"` will fix your problem.
Try using: `schema.default="default"`.

**Reply by MahatiPV** [reply]
Thanks for the inputs @porunov!
So the behavior we see with these 2 configurations (schema.default=none and schema.constraints=true) in place is, schema is not being modified by the erroneous data but the erroneous data is still getting ingested into the graph.
Is there any way that the graph stops this ingestion completely and throws error?

**Reply by porunov** [reply]
I'm not sure I follow the issue to be fair. With schema.default=none data which violates data type is going to be rejected with the error. It will not be ingested into the graph, but instead you will receive an error. Please, checkout your schema. Maybe you already created wrong schema unintentionally by using schema.default=default previously
